### ISSUE TITLE
Ensure consistency between the convenience scripts and the step-by-step installation guide for deb and rpm packages

### ISSUE THREAD

**Issue by kislerdm** [opentofu/opentofu#914]
Followup on #913.

## Problem

It was [noticed](https://github.com/opentofu/opentofu/issues/913#issuecomment-1824148475) that the [repo configuration](https://packagecloud.io/install/repositories/opentofu/tofu/config_file.repo?os=rpm_any&dist=rpm_any&source=script) used by the [convenience script](https://opentofu.org/docs/intro/install/rpm#installing-using-the-convenience-script) differs from the configuration in the [step-by-step instructions](https://opentofu.org/docs/intro/install/rpm#step-by-step-instructions) for the rpm OpenTofu distribution.

|step-by-step instruction|script|
|:---|:---|
|[opentofu]|[opentofu_tofu]|
|name=opentofu|name=opentofu_tofu|
|gpgcheck=1|gpgcheck=0|
|[opentofu-source]|[opentofu_tofu-source]|
|name=opentofu-source|name=opentofu_tofu-source|
|gpgcheck=1|gpgcheck=0|

## Proposed solution

Align the step-by-step instructions with the convenience scripts for rpm and deb.

_Alternatively_, remove the convenience script installation.

**Comment by Yantrio** [comment]
I think the script conditionally sets the gpgcheck based on if it can do it or not.

I think the step-by-step instructions are valid here.

**Comment by kislerdm** [comment]
@Yantrio you are right, here is the script's logic:

```bash
finalize_yum_repo ()
{
  if [ "$_skip_pygpgme" = 0 ]; then
    echo "Installing pygpgme to verify GPG signatures..."
    yum install -y pygpgme --disablerepo="${repo_config_name}"
    pypgpme_check=`rpm -qa | grep -qw pygpgme`
    if [ "$?" != "0" ]; then
      echo
      echo "WARNING: "
      echo "The pygpgme package could not be installed. This means GPG verification is not possible for any RPM installed on your system. "
      echo "To fix this, add a repository with pygpgme. Usualy, the EPEL repository for your system will have this. "
      echo "More information: https://fedoraproject.org/wiki/EPEL#How_can_I_use_these_extra_packages.3F"
      echo

      # set the repo_gpgcheck option to 0
      sed -i'' 's/repo_gpgcheck=1/repo_gpgcheck=0/' /etc/yum.repos.d/$repo_config_name.repo
    fi
  fi
  ...
}
...
```

Maybe we shall share step-by-step instructions only, similar to how docker do it? WDYT @janosdebugs @Yantrio?

**Comment by janosdebugs** [comment]
@kislerdm I'd wait with this until we solved our little GPG signing issue in #913 and #915. I changed the instructions in #917 to be fully scriptable, we could work off of those to provide a full install script.

**Comment by janosdebugs** [comment]
This has been resolved.
